platforms/jvm/plugins-java-base/src/integTest/groovy/org/gradle/api/plugins/jvm/internal/DefaultJvmFeatureTest.groovy

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.plugins.jvm.internal
+
+import org.gradle.api.InvalidUserDataException
+import org.gradle.api.plugins.JavaBasePlugin
+import org.gradle.api.plugins.JavaPluginExtension
+import org.gradle.api.tasks.SourceSet
+import org.gradle.test.fixtures.AbstractProjectBuilderSpec
+
+class DefaultJvmFeatureTest extends AbstractProjectBuilderSpec {
+    def "can create multiple features in a project"() {
+        given:
+        project.plugins.apply(JavaBasePlugin)
+        def ext = project.getExtensions().getByType(JavaPluginExtension)
+
+        SourceSet one = ext.getSourceSets().create("one")
+        SourceSet two = ext.getSourceSets().create("two")
+
+        when:
+        new DefaultJvmFeature("feature1", one, Collections.emptyList(), project, false, false)
+        new DefaultJvmFeature("feature2", two, Collections.emptyList(), project, false, false)
+
+        then:
+        // TODO: There's no way to get the feature by name, so we'll check that some associated tasks and configurations are created
+        project.tasks.getByName(one.getJarTaskName())
+        project.tasks.getByName(two.getJarTaskName())
+        project.configurations.getByName('oneImplementation')
+        project.configurations.getByName('twoImplementation')
+    }
+
+    // TODO: This test definitely isn't appropriate for Component any more, so I've moved it here.
+    // Should features create the sourcesets they are going to use?  If they are being passed in
+    // like this, this is just testing the SourceSets container, not the feature.
```

Comment:
IMO this entire test method isn't relevant anymore and could be deleted.  The source set container enforces the inability to create multiple source sets using the same name - this is just duplicating that.

However...

When I switch this to use the same `sourceSet` instance no exception is thrown.  I need to investigate this more.  A test should definitely prevent this.

Reply:
Yeah I think the medium term solution is that a feature (or target rather once we add multiple targets to a feature) assumes the responsibility of the source set -- therefore not requiring one to be passed in and avoiding this case

Reply:
Nothing prevents this in the same way that nothing prevents a component being created that reuses a feature used by another component.  Which I suppose is a consequence of our design making these types more like value types.

I wonder if we need to think about validating things like this at the end of the configuration phase?

Reply:
We discussed this one.  This test currently makes no sense as a feature requires a source set during construction.  If/when this changes to a feature creating it's own source set as an internal implementation detail, a test such as this will be needed again, but for now it can be deleted.